src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1092,6 +1178,20 @@ private static bool PropertyAccessorCanBeReferenced(MethodInfo? accessor)
                     return null;
                 }
 
+                if (converterType.GetCompatibleBaseClass(JsonConverterFactoryFullName) != null)
```

Comment:
Thanks, I was looking for this logic in the Emitter.cs diff. I know this predates your PR, but shouldn't this method live in `Emitter.cs`?

Reply:
Ah, I'm assuming it's because we're embedding generated initialization expression as a property in the property generation spec.

Reply:
Yes the parser:emitter abstraction is broken here.

Reply:
Ideally it should be moved to the emitter, perhaps in a later PR.